apps/dashboard/app/models/current_user.rb

```diff
@@ -27,7 +28,23 @@ def initialize
   end
 
   def primary_group
-    @primary_group ||= Etc.getgrgid(gid).name
+    @primary_group ||= Etc.getgrgid(gid)
+  end
+
+  def primary_group_name
+    @primary_group_name ||= primary_group.name
+  end
+
+  def group_names
+    @group_names ||= groups.map(&:name)
+  end
+
+  def groups
+    @groups ||= begin
+      groups = Process.groups
+      groups.delete(primary_group.gid)
+      groups.unshift(primary_group.gid).map { |gid| Etc.getgrgid(gid) }
```

Comment:
What is the reason to delete the `primary_group.gid` then pop it back into the array, but in the first position, to then iterate over and hand back in the array. Is that first position important for what is passed back or for `Etc` somehow?

Reply:
It's just a guarantee on our API that the first item in the list is the users' primary group.  Really nothing to do with Etc, just what how we want to control the return value of this method. Let me add a comment for the same.